base/loading.jl

```diff
@@ -633,7 +633,7 @@ end
 
 cache_file_entry(pkg::PkgId) = joinpath(
     "compiled",
-    "v$(VERSION.major).$(VERSION.minor)",
+    "v$(VERSION.major).$(VERSION.minor)" * (JLOptions().debug_level > 1 ? "-debug" : ""),
```

Comment:
`JLOptions().debug_level > 1` is my best guess for the most robust way to test for `julia-debug`, but others much more knowledgeable may have a different opinion.

Reply:
Specifically, I chose to prefer it to something like:
```
endswith(basename(unsafe_string(Base.JLOptions().julia_bin)),"-debug")
```
so as to be insensitive to a name change of the executable.

Reply:
`JLOptions().debug_level` corresponds to the `-g` option and not whether the binary was build with debug information embedded

Reply:
Yes, the test should be `ccall(:jl_is_debugbuild, Cint, ()) > 0`.